Source/WebCore/contentextensions/ContentExtensionActions.cpp

```diff
@@ -149,49 +156,65 @@ void ModifyHeadersAction::serialize(Vector<uint8_t>& vector) const
 
 ModifyHeadersAction ModifyHeadersAction::deserialize(Span<const uint8_t> span)
 {
-    auto serializedLength = deserializeLength(span, 0);
-    auto requestHeadersLength = deserializeLength(span, sizeof(uint32_t));
-    size_t progress = sizeof(uint32_t) * 2;
+    uint32_t priority = deserializeLength(span, 0);
+    auto serializedLength = deserializeLength(span, sizeof(uint32_t));
+    auto requestHeadersLength = deserializeLength(span, sizeof(uint32_t) * 2);
+    size_t progress = sizeof(uint32_t) * 3;
     Vector<ModifyHeaderInfo> requestHeaders;
-    while (progress < requestHeadersLength + sizeof(uint32_t)) {
+    while (progress < requestHeadersLength + sizeof(uint32_t) * 2) {
         auto subspan = span.subspan(progress);
         progress += ModifyHeaderInfo::serializedLength(subspan);
         requestHeaders.append(ModifyHeaderInfo::deserialize(subspan));
     }
-    RELEASE_ASSERT(progress == requestHeadersLength + sizeof(uint32_t));
+
+    RELEASE_ASSERT(progress == requestHeadersLength + sizeof(uint32_t) * 2);
     Vector<ModifyHeaderInfo> responseHeaders;
-    while (progress < serializedLength) {
+    while (progress < serializedLength + sizeof(uint32_t)) {
         auto subspan = span.subspan(progress);
         progress += ModifyHeaderInfo::serializedLength(subspan);
         responseHeaders.append(ModifyHeaderInfo::deserialize(subspan));
     }
-    RELEASE_ASSERT(progress == serializedLength);
-    return { WTFMove(requestHeaders), WTFMove(responseHeaders) };
+    return { WTFMove(requestHeaders), WTFMove(responseHeaders), priority };
 }
 
 size_t ModifyHeadersAction::serializedLength(Span<const uint8_t> span)
 {
     return deserializeLength(span, 0);
 }
 
-void ModifyHeadersAction::applyToRequest(ResourceRequest& request)
+void ModifyHeadersAction::applyToRequest(ResourceRequest& request, HashMap<String, String>& headerNameToFirstOperationApplied)
 {
     for (auto& info : requestHeaders)
-        info.applyToRequest(request);
+        info.applyToRequest(request, headerNameToFirstOperationApplied);
 }
 
-void ModifyHeadersAction::ModifyHeaderInfo::applyToRequest(ResourceRequest& request)
+void ModifyHeadersAction::ModifyHeaderInfo::applyToRequest(ResourceRequest& request, HashMap<String, String>& headerNameToFirstOperationApplied)
```

Comment:
Since the value is only one of 3 strings, I don't think this is the right data structure.  Make a struct with a name and 3 members.